### ISSUE TITLE
Folder structure

### ISSUE THREAD

**Issue by BieberleA** [HZDR-FWDF/GLADOS#13]
Is there a reason why the folder is called "include/glados"? Is "include" not sufficient or do you plan to add more folders into "include"?

**Comment by j-stephan** [comment]
**TL;DR**: It is good style.

**Long version**:

``include`` alone would not be sufficient for two reasons:

First, from a programmer's perspective, it encapsulates the library functionality. Consider a program, which makes use of our ``cuda/algorithm.h`` header file. This hypothetical program looks like this:

```
#include <cuda/algorithm.h>
```

Now the programmer wants to use library CAKE which also supplies ``cuda/algorithm.h`` and follows the same directory structure.

```
#include <cuda/algorithm.h> // uh-oh
```

With the current directory structure the problem disappears:

```
#include <glados/cuda/algorithm.h>
#include <cake/cuda/algorithm.h>
```

Second it is my long-term goal to install the (stable) GLADOS header files into ``/usr/include`` or ``/usr/local/include``. If we omitted the ``glados`` directory all subfiles and subdirectories would directly reside in these directories, thus creating ``/usr/include/cuda``, ``/usr/include/pipeline`` etc. From a system administrator's perspective this is a nightmare. (Imagine all libraries doing this!). As long as the whole functionality isn't supplied by single header file (e.g. some global ``glados.h``) -- which we don't want to do -- it is considered good practice to put all those files into a single library directory.

**Comment by BieberleA** [comment]
Thanks a lot for extending my horizon. I think to local :-).

**Comment by j-stephan** [comment]
No problem.
